dao/src/main/java/com/epam/esm/dao/GiftCertificateDao.java

```diff
@@ -0,0 +1,101 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.entity.GiftCertificate;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This interface represents an api to interact with the GiftCertificate entity in database.
+ * Implementations : {@link com.epam.esm.dao.impl.GiftCertificateDaoImpl} classes.
+ *
+ * @since 1.0
+ */
+public interface GiftCertificateDao {
+
+    /**
+     * This method saves GiftCertificate entity.
+     *
+     * @param certificate GiftCertificate entity without id.
+     * @return GiftCertificate entity with generated id.
+     * @since 1.0
+     */
+    GiftCertificate save(GiftCertificate certificate);
+
+
+    /**
+     * This method get GiftCertificate entity by id.
+     *
+     * @param id GiftCertificate entity's id.
+     * @return Optional of GiftCertificate entity.
+     * @since 1.0
+     */
+    Optional<GiftCertificate> getById(long id);
```

Comment:
Why do you have in get...() in some places and find...() in other?

Reply:
commit 96876a6